sockeye/train.py

```diff
@@ -90,8 +90,8 @@ def check_arg_compatibility(args: argparse.Namespace):
             adjusted_transformer_encoder_model_size = args.num_embed[0] + total_source_factor_size
             check_condition(adjusted_transformer_encoder_model_size % 2 == 0 and
                             adjusted_transformer_encoder_model_size % args.transformer_attention_heads == 0,
-                            "Sum of source factor sizes (%d) has to be a multiple of attention heads (%d)" % (
-                                total_source_factor_size, args.transformer_attention_heads))
+                            "Sum of source factor sizes (%d) has to be even and a multiple of attention heads (%d)" % (
```

Comment:
I'm not sure this error message would be clear to a user. Maybe mention the embedding size "The source embedding size plus all source factor embeddings sizes (%d) has to be even and a ...".

Reply:
Changed it to make it clearer AND to reflect the fact that we regard the source surface form as the first & primary factor:
Sum of source factor sizes, i.e. num-embed plus source-factors-num-embed, (%d) has to be even and a multiple of attention heads (%d)